src/scene/triangle.cpp

```diff
@@ -1,13 +1,13 @@
 #include "triangle.h"
 
-Triangle::Triangle(glm::vec3 p0, glm::vec3 p1, glm::vec3 p2)
-    : mP0(p0), mP1(p1), mP2(p2) {
+Triangle::Triangle(glm::vec3 p0, glm::vec3 p1, glm::vec3 p2, unsigned int id)
```

Comment:
I've been wondering how we could more tightly couple  triangle id to Scene's mTriangleList size and I think  what we could do instead is make addTriangle a friend method of the triangle class like so: 
```c++
class Triangle {
public:
friend void Scene::addTriangle(glm::vec3 p0, glm::vec3 p1, glm::vec3 p2, glm::vec3 color);
};```

we would then have either a private constructor or a private setter for id.

Reply:
So I tried this and can't get it to work because Scene isn't defined yet.
I tried using forward declaration but it says 'incomplete type is not allowed' and I can't include scene.h into triangle.h because that would cause a circular dependency. I can make Scene a friend class of triangle but that will give all of scene's function's the ability to edit a triangle which I don't know if we want. I'm going to push my changes anyway and let you take a look at them. You can maybe pull down this branch and try this for yourself just incase I'm missing something.

Reply:
I'm not going to hold this PR up for, this one thing. Everything LGTM, thanks for making the requested changes.